src/views/primary-auth/CustomButtons.js

```diff
@@ -56,21 +55,19 @@ define([
     },
 
     _createButton: function (options) {
-      var type = options.type.toLowerCase(),
-          attr = 'social-auth-' + type + '-button';
-
+      if(options.isSocialAuth) {
+        options.click =  function(e) {
```

Comment:
It's difficult to have this in Settings.js because the callback references other classes as well. See OAuth2Util.